### PR TITLE
sysding: fix wrong test with grep (grep returns 0 if pattern is found)

### PR DESCRIPTION
**Pull request by grueni**
The return code of grep or egrep is 0 if the pattern is found.
grep -c print the number of lines matching the pattern.

### CODE REVIEW

File: sysding [hunk 1]
```diff
@@ -254,8 +253,7 @@ setup_route() {
 
    # https://github.com/olbohlen/sysding/issues/1
    # check if def_addr is IPv6 and force the -inet6 flag for route(1M)
-   echo ${def_dst} | egrep ":" >/dev/null 2>&1
-   if [ $? -gt 0 ]; then
+   if [ $(echo ${def_dst} | grep -c ":") -gt 0 ]; then
```
Comment (pyhalov) [inline sysding]: please, also fix comment one line above "check if df_dst"...

**Comment by pyhalov** [comment]
Note, that after this PR is merged, you'll have to update oi-userland component, as it fetches specific commit from this repository.

**Comment by grueni** [comment]
Tested with a zone which gets the changed script before the first boot.

**Comment by olbohlen** [comment]
The right thing would be fixing the if [ $? -gt 0 ] to if [ $? -eq 0 ] instead of counting lines.
egrep will return non-zero if it has not matched.

**Comment by olbohlen** [comment]
oh, and also please keep those vim lines out.

**Comment by pyhalov** [comment]
please, create separate PR for create-ipmp fix

**Comment by grueni** [comment]
I gues there is more than one "right" way.
The way choosen is functional equivalent  and leads to the same result.
I choose this type implementation to have the same type of check (-gt 0) as in other places of the script.
But I don't insist on my implementation and  change the test condition.

**Comment by grueni** [comment]
So here the minimal patch.

Which are the advantages of using this:
 echo ${def_dst} | egrep ":" >/dev/null 2>&1
  if [ $? -eq 0 ]; then
versus this
 if [ $(echo ${def_dst} | grep -c ":") -gt 0 ]; then

IMHO the latter is shorter and simpler.

File: sysding [hunk 2]
```diff
@@ -255,7 +255,7 @@ setup_route() {
    # https://github.com/olbohlen/sysding/issues/1
    # check if def_addr is IPv6 and force the -inet6 flag for route(1M)
    echo ${def_dst} | egrep ":" >/dev/null 2>&1
-   if [ $? -gt 0 ]; then
+   if [ $? -eq 0 ]; then
```
Comment (pyhalov) [inline sysding]: please, fix comment above

**Comment by grueni** [comment]
comments changed to current PR

File: sysding [hunk 3]
```diff
@@ -252,10 +252,10 @@ setup_route() {
 
    rt=0
 
-   # https://github.com/olbohlen/sysding/issues/1
+   # https://github.com/OpenIndiana/sysding/pull/5
    # check if def_addr is IPv6 and force the -inet6 flag for route(1M)
```
Comment (pyhalov) [inline sysding]: I meant this comment - s/def_addr/def_dst/

**Comment by olbohlen** [comment]
in your grep, you also should send the output to /dev/null, as it will result in unwanted output.
What if construct we will use is not that important to me. The bug itself is that I wrote -gt instead of -eq.

**Pull request merged**